modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -235,6 +237,26 @@ public CPDefinition addCPDefinition(
 		CProduct cProduct = _cProductLocalService.addCProduct(
 			externalReferenceCode, groupId, userId, new ServiceContext());
 
+		CPSubscriptionType cpSubscriptionType =
+			_cpSubscriptionTypeRegistry.getCPSubscriptionType(subscriptionType);
+
+		if (cpSubscriptionType != null) {
+			subscriptionTypeSettingsUnicodeProperties =
+				cpSubscriptionType.validateSubscriptionTypeSettingsProperties(
+					subscriptionTypeSettingsUnicodeProperties);
+		}
+
+		CPSubscriptionType deliveryCPSubscriptionType =
+			_cpSubscriptionTypeRegistry.getCPSubscriptionType(
+				deliverySubscriptionType);
+
+		if (deliveryCPSubscriptionType != null) {
+			deliverySubscriptionTypeSettingsUnicodeProperties =
+				deliveryCPSubscriptionType.
+					validateDeliverySubscriptionTypeSettingsProperties(
+						deliverySubscriptionTypeSettingsUnicodeProperties);
+		}
```

Comment:
Make this a private method _validateSubscriptionTypeSettings(...) and call it in the update as well